File: src/Omnia.ProcessManagement.Web/client/fx/models/data/drawingdefinitions/DrawingShape.ts
import { GuidValue, MultilingualString } from '@omnia/fx-models';
import { ShapeObject } from '../../../processshape';

export enum DrawingShapeTypes {
    Undefined = 0,
    ProcessStep = 1,
    CustomLink = 2,
    ExternalProcessStep = 3
}

export interface DrawingShape {
    id: GuidValue,
    type: DrawingShapeTypes,
    shape: ShapeObject;
    title: MultilingualString;
}

File: src/Omnia.ProcessManagement.Models/CanvasDefinitions/DrawingShape.cs
using Omnia.ProcessManagement.Models.FabricShape;
using Omnia.ProcessManagement.Models.Shapes;
using System;

namespace Omnia.ProcessManagement.Models.CanvasDefinitions
{
    public enum DrawingShapeTypes
    {
        Undefined = 0,
        ProcessStep = 1,
        CustomLink = 2,
        ExternalProcess = 3
    }

    public class DrawingShape : Omnia.Fx.Models.JsonTypes.OmniaJsonBase
    {
        public Guid Id { get; set; }

        public virtual DrawingShapeTypes Type { get; set; }

        public ShapeObject Shape { get; set; } //TODO - Do we need to have server-side model for this data ?!
    }
}
